### PR TITLE
update for Linux users

### PR DESCRIPTION
**Pull request by mattlub**
related #14 #25 

- [x] add user and password to db connection options
- [x] add instructions in README

**Could someone running linux (ubuntu) verify this works please :D (I haven't actually tested it)**

N.B. no instructions for Windows, but I assume we don't promise to support windows.

### CODE REVIEW

**Comment by eliascodes** [comment]
As mentioned in #14 and foundersandcoders/master-reference#829, I hope this will be made unnecessary by removing the DB entirely. Will leave the PR open for now, but close once those issues are resolved.

**Comment by jamiecoe** [comment]
Why don't we get everyone to make a new user, rather than just the linux people? Could be good practice for SQL commands 🤔

**Comment by m4v15** [comment]
@eliascodes @jamiecoe @sophielevens (as London mentors who will delivering this first)

Where do we want to go with this? As @eliascodes says above, this PR would ideally be superceded by 
https://github.com/foundersandcoders/master-reference/issues/829 : 
>- [ ] Remove database from workshop (substitute with in-memory data-store). Potentially needs guidance. [This Issue](https://github.com/foundersandcoders/express-and-testing-workshop/issues/25) is somewhat relevant and a PR is open to add instructions for linux users. However, this workshop doesn't really _need_ a complicated DB set up so it will probably be better of if it can be replaced with some sort of basic data store (e.g. just a simple object that has the data). Will require rewriting all the queries and db-build stuff.

But that is _quite_ a lot of work which may or may not get done and atm this workshop still breaks for Linux users. I would probably be keen for merging this now (possibly with a change to what @jamiecoe has said above) as without it this WS is annoying for linux users. Thoughts?

**Comment by eliascodes** [comment]
@jamiecoe has contributed his thoughts [here](https://github.com/foundersandcoders/express-and-testing-workshop/issues/14#issuecomment-370837080)

Isn't the real issue here down to configuration? The reason macOS is different is because installing postgres with Homebrew automatically sets the `pg_hba.conf` file to trust all connections from `localhost`. Installing on Linux with `apt-get` doesn't do this, you end up with `peer` or `md5` auth method instead.

When going through the install instructions, who actually got to the [user permissions](https://github.com/macintoshhelper/learn-sql/blob/master/postgresql/setup.md#user-permissions) part, specifically the bit marked "Alternative to avoid using `sudo -u postgres`:"? It explains how to change the `pg_hba.conf` file.

Can someone using Linux test if changing the conf fixes this issue please?

**Comment by m4v15** [comment]
Cool, I'm just gonna quote it here so we have all the discussion in one place: >@eliascodes I disagree with removing the DB from this exercise. This was an important reference point for FAC-12 when using the pg-promisemodule. Also, a valuable reminder-exercise for creating a new DB on your local machine.

Would it be possible to refactor dbConnection.js so the localDB includes user and password fields? Then we include additional instructions in Readme, telling students to add their username and password values to these fields.

Students can check what users they have available with \du command, or create a new user with:
CREATE USER user WITH PASSWORD 'password';

**Comment by m4v15** [comment]
Cool, posting here so all of the discussion is in one place: 

@jamiecoe:
>@eliascodes I disagree with removing the DB from this exercise. This was an important reference point for FAC-12 when using the pg-promisemodule. Also, a valuable reminder-exercise for creating a new DB on your local machine.
>Would it be possible to refactor dbConnection.js so the localDB includes user and password fields? Then we include additional instructions in Readme, telling students to add their username and password values to these fields.
>Students can check what users they have available with \du command, or create a new user with:
CREATE USER user WITH PASSWORD 'password';

**Comment by mattlub** [comment]
@jamiecoe/anyone else, feel free to edit my PR if needs be

**Comment by jamiecoe** [comment]
@eliascodes While we're waiting for a Linux person to test if changing the conf fixes this issue, shall I update the PR to instruct everyone to create a new `user` and `password`?

**Comment by m4v15** [comment]
(that's me saying yes, but also that we should wait for @eliascodes to also agree)

**Comment by eliascodes** [comment]
@jamiecoe 👍

**Comment by mattlub** [comment]
I don't think it's reasonable to expect students to understand about the permissions. But if it is easier and has no drawbacks then editing the `pg_hba.conf` file could/should be in the pg install instructions (not in an extra part), and not here imo

**Comment by matthewdking** [comment]
These instructions to edit `pg_hba.conf` are within the same install instructions but I think most students don't get to/skip the releveant [user permission](https://github.com/macintoshhelper/learn-sql/blob/master/postgresql/setup.md#user-permissions) section.

**Comment by mattlub** [comment]
@matthewdking yeah I mean, you can see why no one gets to that section. If we are going to recommend that it has to come before the 'Congratulations! You've got PostgreSQL setup on Linux (hopefully).' line

**Comment by mattlub** [comment]
basically we have to make these workshops assuming everyone has the same setup.

**Comment by jamiecoe** [comment]
@eliascodes @m4v15 Sorry I know this is late! I've just updated the db instructions to be generic for mac and linux users. I guess this might have to wait till next time.

File: README.md [hunk 1]
```diff
@@ -67,8 +69,11 @@ app.use('/api/v1/', routes)
 
 ## Tasks
 * `git clone` this repository, run `npm install`.
-*  Run `createdb fac-express` - **IMPORTANT**
-*  Setup the database by running `npm run build:db` - **IMPORTANT**
+* Set up the database:
+  * Run `createdb fac-express`
+  * Run `psql` or `pgcli` to enter the pg interactive terminal.
+  * Enter `CREATE USER super WITH SUPERUSER PASSWORD 'password';`
+  * Run `npm run build:db`
```
Comment (m4v15) [inline README.md]: I think we probably need to add something about getting out of psql before running the npm script: 
>Enter `\q` in psql to leave the psql interactive terminal

File: package.json [hunk 2]
```diff
@@ -5,8 +5,7 @@
   "main": "server.js",
   "scripts": {
     "start": "nodemon ./server/",
-    "test": "node ./server/tests/routes.test.js | tap-spec",
-    "test:watch": "tape -w",
```
Comment (m4v15) [inline package.json]: Any reason you got rid of this? Doesn't really matter either way but I'd leave it in just for a reference even if it's not explicitly used. Not too bother tho tbf.

Comment (eliascodes) [reply package.json]: It doesn't work

Comment (jamiecoe) [reply package.json]: Yeah, the `-w` flag doesn't work. You'd need to use the [tape-watch](https://www.npmjs.com/package/tape-watch) node module

File: README.md (hunk 1 above)
Comment (jamiecoe) [reply README.md]: All done 👍

**Comment by m4v15** [comment]
🎉

**Pull request merged**